includes/author-filter/class-author-filter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Adds a filter by author to the Posts page in the Admin dashboard.
+ *
+ * @package Newspack
+ */
+
+/**
+ * Author Filter class
+ */
+class Author_Filter {
+
+	/**
+	 * Flag to make sure hooks are initialized only once
+	 *
+	 * @var boolean
+	 */
+	private static $initiated = false;
+
+	/**
+	 * Initializes the hook
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		if ( ! self::$initiated ) {
+			add_action( 'restrict_manage_posts', array( __CLASS__, 'author_filters_admin' ) );
+			self::$initiated = true;
+		}
+	}
+
+	/**
+	 * Echoes a User Dropdown to be added to the Filters
+	 *
+	 * @param string $post_type The post type the action is acting on.
+	 * @return void
+	 */
+	public static function author_filters_admin( $post_type ) {
+
+		$excluded_post_types = [ 'attachment', 'revision', 'nav_menu_item' ];
+
+		if ( in_array( $post_type, $excluded_post_types, true ) ) {
+			return;
+		}
+
+		$type_object = get_post_type_object( $post_type );
+
+		if ( ! $type_object instanceof WP_Post_Type ) {
+			return;
+		}
+
+		$capability = $type_object->cap->edit_posts;
+
+		$args = array(
+			'show_option_all'  => __( 'All Users', 'newspack' ),
+			'orderby'          => 'display_name',
+			'order'            => 'ASC',
+			'name'             => 'author',
+			'capability'       => [ $capability ], // only users who can edit posts of this post type.
+			'include_selected' => true,
+			'exclude'          => [ get_current_user_id() ], // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn_exclude
```

Comment:
What's the reason behind exclusion of the current user? I think filtering for one's own posts is a valid use case.

Reply:
I'm not removing the current user from the list anymore and it plays nicely with the "Mine" tab. When you click "Mine", your user will be selected in that filter, so it is consistent.

Reply:
Only when Co Authors Plus is active that it will not play nicely and I removed the whole filter when you click "Mine". Otherwise, the filter on the Dropdown would override the "Mine" selection, but "Mine" would still be seen in bold as it it was selected.

(That's because with CAP we change the filter to look at author_name instead of author (ID). But the author filter is still present in the query string, making "Mine" look like it is selected, while the author_name filter will override the author one)